### PR TITLE
dev -> staging (v5.1.0)

### PR DESCRIPTION
**Pull request by praveek**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: AEPLifecycle/Tests/FunctionalTests/LifecycleFunctionalTests.swift [hunk 1]
```diff
@@ -337,6 +337,47 @@ class LifecycleFunctionalTests: XCTestCase {
         XCTAssertEqual(startEvent2.timestamp.timeIntervalSince1970, storedUpgradeDate?.timeIntervalSince1970)
         XCTAssertNotEqual(storedInstallDate, storedUpgradeDate)
     }
+    
+    /// Tests upgrade event when app version number changes
+    func testUpgradeWhenVersionNumberChanged() {
```
Comment (emdobrin) [inline AEPLifecycle/Tests/FunctionalTests/LifecycleFunctionalTests.swift:342]: I find this test quite confusing as it does not clearly state the expectation. can we update to something like this
```
// tests upgrade event is not triggered when app version number changes (it should only trigger on build number change)
testUpgrade_WhenVersionNumberChanged_doesNotTriggerNewUpgrade

File: AEPSignal/Sources/SignalHitProcessor.swift [hunk 2]
```diff
@@ -70,6 +70,10 @@ class SignalHitProcessor: HitProcessing {
             // retry this hit later
             Log.debug(label: LOG_TAG, "Signal request failed with recoverable error \(connection.error?.localizedDescription ?? "") and status code \(connection.responseCode ?? -1). Will retry sending the request later: \(hit.url.absoluteString)")
             completion(false)
+        } else if let error = connection.error as? URLError, error.isRecoverable {
+            // retry this hit later as the request failed with a recoverable transport error
+            Log.debug(label: LOG_TAG, "Signal request failed with unrecoverable error \(connection.error?.localizedDescription ?? "") and status code \(connection.responseCode ?? -1). It will be dropped from the queue: \(urlString)")
```
Comment (emdobrin) [inline AEPSignal/Sources/SignalHitProcessor.swift:75]: this log is inaccurate, the hit is retried in the case of a recoverable error code

**Review by emdobrin — APPROVED** [review]
lg with a small comment on testing

File: AEPLifecycle/Tests/FunctionalTests/LifecycleFunctionalTests.swift (hunk 1 above)
Comment (praveek) [reply AEPLifecycle/Tests/FunctionalTests/LifecycleFunctionalTests.swift:342]: The newly added tests follow the same naming pattern as existing tests. I will work on updating them separately.

**Pull request merged**